src/test/ui/impl-trait/nested-return-type2-tait3.rs

```diff
@@ -24,8 +24,7 @@ type Traitable = impl Trait<Assoc = impl Send>;
 // type does not implement `Duh`, even if its hidden type does. So we error out.
 fn foo() -> Traitable {
     || 42
-    //~^ ERROR `impl Send: Duh` is not satisfied
+    //~^ ERROR the trait bound `Traitable::{opaque#1}: Duh` is not satisfied
```

Comment:
The path here is wrong. `impl Send` should be `Traitable::{opaque#0}::{opaque#0}`.

Reply:
I'm not sure why this happen but I guess we need to call with_hir_id_owner and wrap some of the lowering code that's not currently wrapped?.